### DISCUSSION TITLE
Right way to register events

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#1783] [Help]
```
// first option
    val eventNode: EventNode<EntityEvent> = EventNode.value(
        "pvp listener",
        EventFilter.ENTITY
    ) { obj -> obj.instance == InstanceManager.eventInstance && EventManager.getEventState() == EventStatus.active }

    // second option
    val eventNode: EventNode<EntityEvent> = EventNode.value(
        "pvp listener",
        EventFilter.ENTITY
    ) { obj -> isEvent(obj) }

    private fun isEvent(entity: Entity) : Boolean {
        return (entity.instance == InstanceManager.eventInstance && EventManager.getEventState() == EventStatus.active)
    }
```

So if I'm right, the first option is really bad and will result in performance issues, and second option is fine, and don't result in performace issues? (I just want the confirmation)

**Comment by Tofaa2** [comment]
Im so confused on what you're trying to do but if you're testing performance you can use Intellij's profiler and try both methods

**Reply by DevBlokJochem** [reply]
I just want to know which method is better for performance.

**Reply by Tofaa2** [reply]
try it in a profiler

**Comment by KrystilizeNevaDies** [comment]
No these will both have almost identical performance, in general you don't need to worry about performance though, wait until you actually have features in your gamemode.

**Reply by DevBlokJochem** [reply]
I'm planning to do a pvp event with around the 500 players. I have playermove events etc. It isn't that mutch work to write the Predicate in the right way. Which method do you use?
